### DISCUSSION TITLE
How to use Heimdall with MqSQL, MariaDB or PostgreSQL?

### DISCUSSION THREAD

**Discussion by keriati** [linuxserver/Heimdall#945] [Q&A]
From version 2.4.16 on, Heimdall should work with MySQL, MariaDB and PosgreSQL databases without issues.

To use MySQL, MariaDB or PostgreSQL databases instead of `sqlite` add your databse details to your .env file like:

    DB_CONNECTION=<mysql or pgsql>
    DB_HOST=<hostname/ip>
    DB_PORT=<port>
    DB_DATABASE=<database>
    DB_USERNAME=<username>
    DB_PASSWORD=<password>

**Comment by w-marco** [comment]
Just to add, for anyone running this in docker, because it could become confusing:

This does not need to be set as docker environment variables, but rather inside the .env file that's inside the 'www' folder wherever you mount your /config path from. 

So you need to put these values into `/PATH/TO/HEIMDALL/DATA/www/.env`. Setting them as docker environment variables will create the tables but it will not make the application actually write stuff into the DB.
